src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2934,7 +2934,8 @@ public IEnumerable<ISymbol> GetSymbolsWithName(Func<string, bool> predicate, Sym
                 var result = new HashSet<ISymbol>();
                 var spine = new List<MergedNamespaceOrTypeDeclaration>();
 
-                AppendSymbolsWithName(spine, _compilation.Declarations.MergedRoot, predicate, filter, result, cancellationToken);
+                var mergedRoot = ((SourceNamespaceSymbol)_compilation.SourceModule.GlobalNamespace).MergedDeclaration;
```

Comment:
Definitely interested in knowing the answer to this.  In particular the potential impact this could have to performance.